circle.yml

```diff
@@ -32,3 +32,12 @@ jobs:
           if [[ $TAG =~ ^[0-9]+(\.[0-9]+)+$ ]]; then
             ./gradlew -i bintrayUpload
           fi
+workflows:
+  version: 2
+  build-and-deploy:
+    jobs:
+      - build:
+          filters:
+            # run on all branches and tags
+            tags:
+              only: /.*/
```

Comment:
This works as a quick-and-dirty fix, but isn't ideal because you can get caught in the case where, if you push a commit that's tagged, you can get 2 builds that will both attempt to run the publish action (because on both the tag build and the main build `git describe --tags` will return the version tag).

Since we're using workflows anyway, the more correct way to structure this is to remove the `deploy` step from the `build` job, create a new `publish` job, and then set the workflows such that `build` always runs, `publish` depends on `build` and `publish` only runs on matching tags (these features weren't available when I first wrote this CircleCI 2 file).

Also, we should move the CircleCI file to `.circleci/config.yml`.